util/webhook_grip.go

```diff
@@ -166,21 +166,28 @@ func (w *evergreenWebhookLogger) send(m message.Composer) error {
 		if err != nil {
 			return true, errors.Wrap(err, "sending webhook data")
 		}
-		if resp.StatusCode < 200 || resp.StatusCode >= 300 {
-			return true, errors.Errorf("response was %d (%s)", resp.StatusCode, http.StatusText(resp.StatusCode))
-		}
 
 		body, err := io.ReadAll(resp.Body)
 		if err != nil {
 			return true, errors.Wrap(err, "reading webhook response")
```

Comment:
Do we want to log in this error case as well?

Reply:
Hmm yeah I guess we could. I consolidated the logic for logging in the error/success cases.